src/Analyzers/CSharp/Analyzers/InlineDeclaration/CSharpInlineDeclarationDiagnosticAnalyzer.cs

```diff
@@ -52,8 +52,9 @@ protected override void InitializeWorker(AnalysisContext context)
             {
                 var compilation = compilationContext.Compilation;
                 var expressionType = compilation.GetTypeByMetadataName(typeof(Expression<>).FullName!);
-                compilationContext.RegisterSyntaxNodeAction(
-                    syntaxContext => AnalyzeSyntaxNode(syntaxContext, expressionType), SyntaxKind.Argument);
+                compilationContext.RegisterCodeBlockStartAction<SyntaxKind>(blockStartContext =>
+                    blockStartContext.RegisterSyntaxNodeAction(
+                        syntaxContext => AnalyzeSyntaxNode(syntaxContext, expressionType), SyntaxKind.Argument));
```

Comment:
i would doc these cases.  something like "we analyze arguments, but we report diagnostics on a higher-up node.  So we register a code-block action to allow us to blah blah blah."

Reply:
Sure. Note that I confirmed that unit tests fail if this nested registration is not done after the test framework changes. Will add comments as well.

Reply:
yup.  this is more for maintenance purposes.